File: reforged_16/src/main/java/top/figsq/poketip/poketip/impl/PokemonWrapper.java
package top.figsq.poketip.poketip.impl;

import com.pixelmonmod.pixelmon.api.pokemon.Pokemon;
import com.pixelmonmod.pixelmon.api.pokemon.species.Species;
import lombok.Getter;
import top.figsq.poketip.poketip.PokeTip;
import top.figsq.poketip.poketip.api.pokemon.IPokemonWrapper;
import top.figsq.poketip.poketip.api.pokemon.ISpeciesWrapper;

@Getter
public class PokemonWrapper extends IPokemonWrapper<Pokemon> {
    private final Pokemon original;

    public PokemonWrapper(Pokemon original){
        this.original = original;
    }

    @Override
    public ISpeciesWrapper<Species> getSpecies() {
        return ((PokeTip) PokeTip.getInstance()).getSpeciesWrapperFactory().create(original.getSpecies());
    }

    @Override
    public String getName() {
        return this.original.getLocalizedName();
    }

    @Override
    public Class<Pokemon> getType() {
        return Pokemon.class;
    }
}

File: cobblemon_21/src/main/kotlin/top/figsq/poketip/poketip/impl/PokemonWrapper.kt
package top.figsq.poketip.poketip.impl

import com.cobblemon.mod.common.pokemon.Pokemon
import com.cobblemon.mod.common.pokemon.Species
import lombok.Getter
import top.figsq.poketip.poketip.PokeTip
import top.figsq.poketip.poketip.api.pokemon.IPokemonWrapper
import top.figsq.poketip.poketip.api.pokemon.ISpeciesWrapper

class PokemonWrapper(private val original: Pokemon) : IPokemonWrapper<Pokemon>() {
    override fun getSpecies(): ISpeciesWrapper<Species> {
        return PokeTip.INSTANCE.speciesWrapperFactory.create(original.species)
    }

    override fun getName(): String {
        return this.original.species.translatedName.string
    }

    override fun getOriginal(): Pokemon {
        return this.original
    }

    override fun getType(): Class<Pokemon> {
        return Pokemon::class.java
    }
}
